types/config.d.ts

```diff
@@ -161,14 +161,14 @@ declare global {
         artifacts: ArtifactDefn[];
       }
 
-      export interface ArtifactDefn {
+      export interface ArtifactDefn<TDependencies extends Gatherer.DependencyKey = Gatherer.DependencyKey> {
```

Comment:
this defaults to "all artifacts" to signal that you need some artifacts to pass in as deps, but unclear yet how much casting vs. validation we'll be able to do dynamicly